block/produce.go

```diff
@@ -96,18 +100,26 @@ func (m *Manager) ProduceBlockLoop(ctx context.Context, bytesProducedC chan int)
 	}
 }
 
+// nextProposerInfo holds information about the next proposer.
+type nextProposerInfo struct {
```

Comment:
this `nextProposerInfo struct` is uneeded, and changes alot of APIs just to be used in `consensusMsgsOnCreateBlock`.
`nextProposerAddr` can be queried using `m.State.Sequencers.GetByHash` (similar to `SetByHash`)